src/test/java/org/folio/circulation/resources/renewal/OverrideRenewalStrategyTest.java

```diff
@@ -0,0 +1,361 @@
+      .withLoanable(false)
+      .create();
+
+    final Result<Loan> renewedLoan = renew(LoanPolicy.from(loanPolicyJson), overrideDate);
+
+    assertDueDate(overrideDate, renewedLoan);
+  }
+
+  @Test
+  public void shouldUseOverrideDateWhenLoanIsNotRenewable() {
+    final DateTime overrideDate = now().plusMonths(1);
+    final JsonObject loanPolicyJson = new LoanPolicyBuilder()
+      .notRenewable()
+      .create();
+
+    final Result<Loan> renewedLoan = renew(LoanPolicy.from(loanPolicyJson), overrideDate);
+
+    assertDueDate(overrideDate, renewedLoan);
+  }
+
+  @Test
+  public void overrideDateIsRequiredWhenLoanIsNotRenewable() {
+    final JsonObject loanPolicyJson = new LoanPolicyBuilder()
+      .notRenewable()
+      .create();
+
+    final Result<Loan> renewedLoan = renew(LoanPolicy.from(loanPolicyJson), null);
+
+    assertValidationError(renewedLoan,
+      "New due date must be specified when due date calculation fails");
+  }
+
+  @Test
+  public void overrideDateIsRequiredWhenLoanIsNotLoanable() {
+    final JsonObject loanPolicyJson = new LoanPolicyBuilder()
+      .withLoanable(false)
+      .create();
+
+    final Result<Loan> renewedLoan = renew(LoanPolicy.from(loanPolicyJson), null);
+
+    assertValidationError(renewedLoan,
+      "New due date must be specified when due date calculation fails");
+  }
+
+  @Test
+  public void shouldUseOverrideDateWhenUnableToCalculateProposedDueDate() {
+    final DateTime overrideDate = now().plusMonths(1);
+    final JsonObject loanPolicyJson = rollingPolicy().create();
+
+    // Use undefined strategy to break due date calculation
+    writeByPath(loanPolicyJson, "UNDEFINED_STRATEGY", "renewalsPolicy", "renewFromId");
+
+    final Result<Loan> renewedLoan = renew(LoanPolicy.from(loanPolicyJson), overrideDate);
+
+    assertDueDate(overrideDate, renewedLoan);
+  }
+
+  @Test
+  public void overrideDateIsRequiredWhenUnableToCalculateProposedDueDate() {
+    final JsonObject loanPolicyJson = rollingPolicy().create();
+
+    writeByPath(loanPolicyJson, "UNDEFINED_STRATEGY", "renewalsPolicy", "renewFromId");
+
+    final Result<Loan> renewedLoan = renew(LoanPolicy.from(loanPolicyJson), null);
+
+    assertValidationError(renewedLoan,
+      "New due date must be specified when due date calculation fails");
+  }
+
+  @Test
+  public void shouldUseOverrideDateWhenReachedNumberOfRenewalsAndNewDueDateBeforeCurrent() {
+    final DateTime overrideDueDate = now(UTC).plusWeeks(2);
+    final LoanPolicy loanPolicy = LoanPolicy.from(rollingPolicy()
+      .limitedRenewals(1)
+      .create());
+
+    final Loan loan = Loan.from(new JsonObject().put("renewalCount", 2))
+      .changeDueDate(now(UTC).plusWeeks(1).plusSeconds(1))
+      .withLoanPolicy(loanPolicy);
+
+    final Result<Loan> renewedLoan = renew(loan, overrideDueDate);
+
+    assertDueDate(overrideDueDate, renewedLoan);
+  }
+
+  @Test
+  public void shouldUseProposedDueDateWhenReachedNumberOfRenewalsAndNewDueDateAfterCurrent() {
```

Comment:
What is the difference between the proposed due date and the override due date, is the proposed the one determined automatically by the system?

Reply:
Yes, the proposed due date is date and time calculated by policy and override due date - due date specified by user in override renew request.

Reply:
I think of proposed as the due date proposed by the client, maybe a term like `calculated` could make this distinction stronger, or alternatively a comment to explain the distinction?

Reply:
Make sense, I use `calculated` term now.